jaxrs/src/main/java/io/airlift/jaxrs/AsyncResponseBinder.java

```diff
@@ -0,0 +1,130 @@
+    private boolean cancelOnClientDisconnect = true;
+    private ListenableFuture<?> future;
+    private Executor httpExecutor;
+    private Optional<Duration> timeout = Optional.empty();
+    private Function<Duration, Supplier<Response>> timeoutResponse = AsyncResponseBinder::defaultTimeoutResponse;
+
+    private AsyncResponseBinder(HttpServletRequest servletRequest)
+    {
+        this.servletRequest = requireNonNull(servletRequest, "servletRequest is null");
+    }
+
+    public AsyncResponseBinder cancelOnClientDisconnection(boolean cancelOnClientDisconnection)
+    {
+        this.cancelOnClientDisconnect = cancelOnClientDisconnection;
+        return this;
+    }
+
+    public AsyncResponseBinder withFuture(ListenableFuture<?> future)
+    {
+        this.future = requireNonNull(future, "future is null");
+        return this;
+    }
+
+    public AsyncResponseBinder withExecutor(Executor httpExecutor)
+    {
+        this.httpExecutor = requireNonNull(httpExecutor, "httpExecutor is null");
+        return this;
+    }
+
+    public AsyncResponseBinder withTimeout(Duration timeout)
+    {
+        this.timeout = Optional.of(timeout);
+        return this;
+    }
+
+    public AsyncResponseBinder withTimeoutResponse(Response response)
+    {
+        requireNonNull(response, "response is null");
+        this.timeoutResponse = (timeout) -> () -> response;
+        return this;
+    }
+
+    public AsyncResponseBinder withTimeoutResponse(Supplier<Response> responseSupplier)
+    {
+        requireNonNull(responseSupplier, "responseSupplier is null");
+        this.timeoutResponse = (timeout) -> responseSupplier;
+        return this;
+    }
+
+    public void bindTo(AsyncResponse response)
+    {
+        AsyncResponseHandler<?> asyncResponseHandler = bindAsyncResponse(response, future, httpExecutor);
+        if (cancelOnClientDisconnect) {
+            servletRequest.getAsyncContext().addListener(new AsyncListener()
+            {
+                @Override
+                public void onComplete(AsyncEvent event)
+                {
+                }
+
+                @Override
+                public void onTimeout(AsyncEvent event)
+                {
+                }
+
+                @Override
+                public void onError(AsyncEvent event)
+                {
+                    // Jetty will signal the client disconnection with the EofException
+                    if (event.getThrowable().getMessage().contains("cancel_stream_error")) {
+                        asyncResponseHandler.clientDisconnected();
+                    }
+                }
+
+                @Override
+                public void onStartAsync(AsyncEvent event)
+                {
+                }
+            });
+        }
+        timeout.ifPresent(timeout -> asyncResponseHandler.withTimeout(timeout, timeoutResponse.apply(timeout)));
+    }
+
+    private static Supplier<Response> defaultTimeoutResponse(Duration timeout)
```

Comment:
Rather than duplicating the code from `AsyncResponseHandler`, just call the `withTimeout(Duration)` variant